src/launcher/java/org/truffleruby/launcher/Launcher.java

```diff
@@ -121,6 +125,19 @@ public static void metricsEnd() {
         printTruffleMemoryMetric();
     }
 
+    private static void debugPreInitialization() {
+        if (DEBUG_PRE_INIT_CONTEXT) {
+            try {
+                final Class<?> holderClz = Class.forName("org.graalvm.polyglot.Engine$ImplHolder");
+                final Method preInitMethod = holderClz.getDeclaredMethod("preInitializeEngine");
+                preInitMethod.setAccessible(true);
+                preInitMethod.invoke(null);
+            } catch (ReflectiveOperationException e) {
+                e.printStackTrace();
+            }
+        }
```

Comment:
What does this method do?

Reply:
It trigger the logic for Context pre-initialization, which is useful to debug it on the JVM (doesn't require building an image every time). As the commit message says, this should eventually move into Truffle.